python/graphstorm/graphpartitioning/array_readwriter/parquet.py

```diff
@@ -0,0 +1,82 @@
+import logging
+
+import numpy as np
+import pandas as pd
+import pyarrow
+import pyarrow.parquet
+
+from .registry import register_array_parser
+
+
+@register_array_parser("parquet")
+class ParquetArrayParser(object):
+    """Wrapper class for parquet read/write operations."""
+    def __init__(self):
+        pass
+
+    def read(self, path):
+        """ Read operation on a file.
+
+        Argument
+        --------
+        path: str
+            Path of a file from which to read.
+        """
+        logging.debug("Reading from %s using parquet format" % path)
+        metadata = pyarrow.parquet.read_metadata(path)
+        metadata = metadata.schema.to_arrow_schema().metadata
+
+        # As parquet data are tabularized, we assume the dim of ndarray is 2.
+        # If not, it should be explictly specified in the file as metadata.
+        if metadata:
+            shape = metadata.get(b"shape", None)
+        else:
+            shape = None
+        table = pyarrow.parquet.read_table(path, memory_map=True)
+
+        data_types = table.schema.types
+        # Spark ML feature processing produces single-column parquet files where each row is a vector object
+        if len(data_types) == 1 and isinstance(data_types[0], pyarrow.ListType):
```

Comment:
this is to store multi-dimension arrays?

Reply:
The else part of this statement is meant for multi-dimensional arrays.